beacon_node/beacon_chain/src/beacon_block_reward.rs

```diff
@@ -214,7 +220,9 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                     if participation_flag_indices.contains(&flag_index)
                         && !validator_participation.has_flag(flag_index)?
                     {
-                        validator_participation.add_flag(flag_index)?;
+                        if set_participation_flag {
+                            validator_participation.add_flag(flag_index)?;
```

Comment:
We need to update this here, because we're also reading from `validator_participation` as part of the `if` condition to work out whether to award a reward here.